File: euler-15.go
package main

import (
	"fmt"
	"time"
)

func start(gridX, gridY int) int {
	var forward func(posX, posY int) int
	forward = func(posX, posY int) int {
		x, y := 0, 0
		if posX < gridX {
			x = forward(posX+1, posY)
		}
		if posY < gridY {
			y = forward(posX, posY+1)
		}

		if x == 0 && y == 0 {
			return 1
		}
		return x + y
	}
	return forward(0, 0)
}

func square(size int) int {
	return start(size, size)
}

func main() {
	for i := 1; i <= 20; i++ {
		startTime := time.Now()
		result := square(i)
		fmt.Printf("Size: %d, Result: %d, Time: %v seconds\n", i, result, time.Since(startTime).Seconds())
	}
	fmt.Println("Done")
}

File: euler_15.py
from time import time

import math

def fact(n): return 1 if n <= 1 else n*fact(n-1)

def bin_coeff(n, k):
    n_fact = fact(n)
    k_fact = fact(k)
    diff_fact = fact(n-k)

    return n_fact / (k_fact * diff_fact)

def number_of_routes(n):
    # The number of routes is the binomial coefficient C(2n, n)
    return bin_coeff(2 * n, n)

def main():
    for i  in range(1, 21):
        start = time()
        print(f'Size: {i}, Result: {number_of_routes(i)}, Time: {time() - start} seconds')
    return 'Done'